src/wiki/plugins/macros/mdx/toc.py

```diff
@@ -6,16 +8,92 @@
+        toc_top = 1
+        toc_bottom = int(toc_depth)
+    return {"toc_top": toc_top, "toc_bottom": toc_bottom}
+
+
+def process_value(org_val, new_val):
+    try:
+        if type(new_val) is str:
+            new_val = new_val.lstrip("'").rstrip("'")
+            if type(org_val) is not str:
+                new_val = json.loads(new_val.lower())
+        elif type(new_val) is not type(org_val):
+            new_val = type(org_val)(new_val)
+    except Exception:
+        return org_val
+    else:
+        return new_val
+
+
 def wiki_slugify(*args, **kwargs):
     return HEADER_ID_PREFIX + slugify(*args, **kwargs)
 
 
 class WikiTreeProcessorClass(TocTreeprocessor):
+    CACHED_KWARGS = dict()  # Used to cache arguments parsed by the MacroPattern
+    # Used to map the keyword arguments to the Class Objects attribute name.
+    TOC_CONFIG_VALUES = {
+        "title": "title",
+        "baselevel": "base_level",
+        "separator": "sep",
+        "anchorlink": "use_anchors",
+        "anchorlink_class": "anchorlink_class",
+        "permalink": "use_permalinks",
+        "permalink_class": "permalink_class",
+        "permalink_title": "permalink_title",
+        "toc_depth": process_toc_depth,
+        "toc_top": "toc_top",
+        "toc_bottom": "toc_bottom",
+    }
+
     def run(self, doc):
         # Necessary because self.title is set to a LazyObject via gettext_lazy
         if self.title:
             self.title = str(self.title)
-        super().run(doc)
+
+        tmp_kwargs = dict()
+        try:
+            # Set config and save defaults to tmp
+            for k, v in WikiTreeProcessorClass.CACHED_KWARGS.items():
+                if k in WikiTreeProcessorClass.TOC_CONFIG_VALUES:
+                    if callable(WikiTreeProcessorClass.TOC_CONFIG_VALUES[k]):
+                        for tock, tocv in WikiTreeProcessorClass.TOC_CONFIG_VALUES[k](
+                            v
+                        ).items():
+                            tmp_kwargs[tock] = getattr(
+                                self, WikiTreeProcessorClass.TOC_CONFIG_VALUES[tock]
+                            )
+                            setattr(
+                                self,
+                                WikiTreeProcessorClass.TOC_CONFIG_VALUES[tock],
+                                process_value(tmp_kwargs[tock], tocv),
+                            )
+                    else:
+                        tmp_kwargs[
+                            WikiTreeProcessorClass.TOC_CONFIG_VALUES[k]
+                        ] = getattr(self, WikiTreeProcessorClass.TOC_CONFIG_VALUES[k])
+                        setattr(
+                            self,
+                            WikiTreeProcessorClass.TOC_CONFIG_VALUES[k],
+                            process_value(
+                                tmp_kwargs[WikiTreeProcessorClass.TOC_CONFIG_VALUES[k]],
+                                v,
+                            ),
+                        )
```

Comment:
I'm not so concerned with the indentation levels, I guess some people would point that out as an issue.

But it'd be nice with some inline comments on the inner-most parts to add a narrative and make it more readable? Would you mind adding that?

Reply:
I will attempt to make this look cleaner while respecting lint settings.

Reply:
I don't wanna stress this change with "stylistic" changes, but just adding some helpful comments could make it easier to read :+1:

Reply:
When Django-Wiki supports the latest version of Python-Markdown this will need to be re-evaluated because there is the possibility of taking in arguments that are structured like a list. 

I think I will do a simple check to see if the string starts with [ and ends with ] and then use some regex to part it into a list. However, at this time, that is sorta out of the scope of this PR. 

I would like to point out though, that I originally took on the Issue: https://github.com/django-wiki/django-wiki/issues/1042 because at the time, I didn't realize that Python-Markdown 3.4 wasn't supported. And I wanted to play around with adjusting the class of the div that wraps the TOC. 

So... hopefully in the future!?